### PR TITLE
EOS-16426 Support ResetPassword for S3 Account

### PR DESCRIPTION
**Pull request by shalakadharap**

### CODE REVIEW

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/405/

**Comment by shalakadharap** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/S3server-7.8.2003/407/

**Review by bkirunge7 — APPROVED** [review]

File: auth/server/src/main/java/com/seagates3/controller/IAMController.java [hunk 1]
```diff
@@ -131,6 +132,98 @@ ServerResponse serve(FullHttpRequest httpRequest,
...
+      catch (InvalidRequestorException ex) {
+        LOGGER.debug(ex.getServerResponse().getResponseBody());
+      }
+      if (requestor == null) {
+        LOGGER.debug(
+            "Validating user on the basis of ldap credentials entered");
+        AccessKey akey = new AccessKey();
+        String ldapUser = AuthServerConfig.getLdapLoginCN();
+        if (!ldapUser.equals(clientRequestToken.getAccessKeyId())) {
+          LOGGER.error("Invalid ldap user, authentication failed");
```
Comment (sachitanands) [inline auth/server/src/main/java/com/seagates3/controller/IAMController.java:174]: can we log the ldapUser value as well in the log

File: auth/server/src/main/java/com/seagates3/controller/IAMController.java [hunk 2]
```diff
@@ -131,6 +132,98 @@ ServerResponse serve(FullHttpRequest httpRequest,
...
+        akey.setSecretKey(AuthServerConfig.getLdapLoginPassword());
+        requestor = new Requestor();
+        requestor.setAccessKey(akey);
+        LOGGER.debug("Calling signature validator.");
+
+        perf.startClock();
+        serverResponse =
+            new SignatureValidator().validate(clientRequestToken, requestor);
+        perf.endClock();
+        perf.printTime("Request validation");
```
Comment (sachitanands) [inline auth/server/src/main/java/com/seagates3/controller/IAMController.java:190]: print msg can be "request validation complete"

**Review by sachitanands — APPROVED** [review]
pls address minor comments .. rest of the changes looks good

File: auth/server/src/main/java/com/seagates3/controller/IAMController.java (hunk 1 above)
Comment (shalakadharap) [reply auth/server/src/main/java/com/seagates3/controller/IAMController.java:174]: we shoud not be logging credentials

File: auth/server/src/main/java/com/seagates3/controller/IAMController.java (hunk 2 above)
Comment (shalakadharap) [reply auth/server/src/main/java/com/seagates3/controller/IAMController.java:190]: its actually performance log related logic. Hence not touching it as of now.

Comment (sachitanands) [reply auth/server/src/main/java/com/seagates3/controller/IAMController.java:190]: okay

**Pull request merged**